### PR TITLE
Remove offset-based pagination from `list_jobs` function in `DatabricksHook`

### PR DESCRIPTION
**Pull request by oleksiidav**
Databricks has deprecated the offset parameter from [ListJobs API](https://docs.databricks.com/api/workspace/jobs/list), and to soften and control the inevitable breaking change for Airflow users, as well as to conceal the implementation details we are removing the offset-based pagination from all connectors.

This is a follow-up to #33472. Airflow users passing offsets to `list_jobs` should be receiving deprecation warnings already starting from [this release](https://github.com/apache/airflow/issues/34930).

### CODE REVIEW

**Comment by oleksiidav** [comment]
@alexott could you please take a look and assign a maintainer to review if you can?

**Comment by eladkal** [comment]
> This is a follow-up to https://github.com/apache/airflow/pull/33472. Airflow users passing offsets to list_jobs have been receiving deprecation warnings for the last month.

https://github.com/apache/airflow/pull/33472 have not been released yet so no has been warned

**Comment by oleksiidav** [comment]
> https://github.com/apache/airflow/pull/33472 have not been released yet so no have been warned

I see @eladkal ! In any case, there is at least a month of time between that PR and this one. So this change can surely go into the following release after [this one](https://github.com/apache/airflow/issues/34930), assuming monthly release cadence

**Comment by Lee-W** [comment]
Do we need to do something like https://github.com/apache/airflow/pull/34738#discussion_r1344665048 in this PR as well? Looks like a breaking change.

**Comment by uranusjr** [comment]
Yes we do. Or to not make this a breaking change, we can continue to accept the argument but just ignore it with a warning. I would prefer dropping it entirely though.

**Comment by oleksiidav** [comment]
Thanks @uranusjr, I have added the breaking change note the changelog

File: airflow/providers/databricks/CHANGELOG.rst [hunk 1]
```diff
@@ -61,6 +61,13 @@ Misc
 4.5.0
 .....
 
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+The ``offset`` parameter has been deprecated from ``list_jobs`` in favor of faster pagination with ``page_token`` similarly to `Databricks API <https://docs.databricks.com/api/workspace/jobs/list>`_.
```
Comment (eladkal) [inline airflow/providers/databricks/CHANGELOG.rst]: It's not in the right place it should be at the top of the log before the
```
Changelog
---------
```

File: airflow/providers/databricks/CHANGELOG.rst [hunk 2]
```diff
@@ -61,6 +61,13 @@ Misc
...
+
+* ``Remove offset-based pagination from list_jobs function in DatabricksHook (#34926)``
+
```
Comment (eladkal) [inline airflow/providers/databricks/CHANGELOG.rst]: Please remove this line.
This will be created automatically by release manager

**Comment by eladkal** [comment]
Please also add `5.0.0` entry to versions in `provider.yaml:`

https://github.com/apache/airflow/blob/e9987d50598f70d84cbb2a5d964e21020e81c080/airflow/providers/databricks/provider.yaml#L25-L28

File: airflow/providers/databricks/CHANGELOG.rst (hunk 1 above)
Comment (oleksiidav) [reply airflow/providers/databricks/CHANGELOG.rst]: Gotcha, fixed!

**Comment by oleksiidav** [comment]
Thanks @eladkal ! Applied changes

**Review by harishkesavarao — APPROVED** [review]
Looks good, agree with removing the `offset` altogether.

**Comment by oleksiidav** [comment]
I have spent some time but I'm really not sure where the bad revision error is coming from in [provider docs GH workflow](https://github.com/apache/airflow/actions/runs/6696355323/job/18194182959?pr=34926). I suspect it's the different commit introducing v4.7.0 in between before I added 5.0.0. @eladkal could I ask you to take over and merge? Or at least provide advice how to solve this?

**Comment by potiuk** [comment]
You need to wait until 4.7.0 is released. It is being voted currently - but not yet released and the error you see is when documentation is generated to show the difference between released 4.7.0 and your 5.0.0 - but since 4.7.0 is not yet released (and tag not set) this is impossible to generate the difference from 4.7.0. 

Just wait until 4.7.0 is released (will be announced on the devlist - likely tomorrow) and rebase. It should get fixed.

**Comment by oleksiidav** [comment]
Ah, thank you very much @potiuk! Will wait for a few days

**Comment by oleksiidav** [comment]
Hi @potiuk @eladkal, let's merge!

**Review by potiuk — APPROVED** [review]

**Pull request merged**